checks/__init__.py

```diff
@@ -502,6 +512,18 @@ def get_service_checks(self):
         self.service_checks = []
         return service_checks
 
+    def get_service_metadata(self):
+        """
+        Return a list of the metadata dictionaries saved by the check -if any-
+        and clears them out of the instance's service_checks list
+
+        @return the list of metadata saved by this check
+        @rtype list of metadata dicts
+        """
+        service_metadata = self.service_metadata
+        self.service_metadata = []
```

Comment:
`self.service_metadata` is indexing metadata by instance. If we have multiple versions on an application running on different instances, we'll end up with `[{'version': 1.0}, {'version': 1.5}]`